pagerduty/resource_pagerduty_event_orchestration_path_router.go

```diff
@@ -1,21 +1,23 @@
 package pagerduty
 
 import (
+	"context"
 	"fmt"
 	"log"
 	"time"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/heimweh/go-pagerduty/pagerduty"
 )
 
 func resourcePagerDutyEventOrchestrationPathRouter() *schema.Resource {
 	return &schema.Resource{
-		Read:   resourcePagerDutyEventOrchestrationPathRouterRead,
-		Create: resourcePagerDutyEventOrchestrationPathRouterCreate,
-		Update: resourcePagerDutyEventOrchestrationPathRouterUpdate,
-		Delete: resourcePagerDutyEventOrchestrationPathRouterDelete,
+		ReadContext:   resourcePagerDutyEventOrchestrationPathRouterRead,
+		CreateContext: resourcePagerDutyEventOrchestrationPathRouterCreate,
+		UpdateContext: resourcePagerDutyEventOrchestrationPathRouterUpdate,
+		DeleteContext: resourcePagerDutyEventOrchestrationPathRouterDelete,
```

Comment:
In order to return warnings to the user we need to switch to using [context-aware methods](https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.24.1/helper/schema/resource.go#L624-L633). These methods return `diag.Diagnostics` instead of `error` or `nil`, which allows us to load warnings returned by the Update API into the `Diagnostics` collection.